pkg/ccl/changefeedccl/sink_pubsub.go

```diff
@@ -506,6 +529,28 @@ func (p *gcpPubsubClient) forEachTopic(f func(name string, topicClient *pubsub.T
 	})
 }
 
+func (p *gcpPubsubClient) recordAutocreateError(e error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.mu.autocreateError = e
+}
+
+func (p *gcpPubsubClient) recordPublishError(e error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.mu.publishError = e
+}
+
+// connectivityError returns any errors encountered while writing to gcp.
+func (p *gcpPubsubClient) connectivityError() error {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	if status.Code(p.mu.publishError) == codes.NotFound && p.mu.autocreateError != nil {
+		return errors.Wrap(p.mu.autocreateError, "Topic not found, and attempt to autocreate it errored")
```

Comment:
perhaps "attempt to autocreate failed"?
Also, should we add a hint re "pubsub editor role"?

Reply:
Done and done, thanks!